overlord/devicestate/devicestate_test.go

```diff
@@ -208,25 +211,15 @@ func (s *deviceMgrBaseSuite) SetUpTest(c *C) {
 	s.restoreCloudInitStatusRestore = devicestate.MockCloudInitStatus(func() (sysconfig.CloudInitState, error) {
 		return sysconfig.CloudInitRestrictedBySnapd, nil
 	})
+	s.AddCleanup(s.restoreCloudInitStatusRestore)
+
+	s.AddCleanup(func() { s.ancillary = nil })
 }
 
 func (s *deviceMgrBaseSuite) newStore(devBE storecontext.DeviceBackend) snapstate.StoreService {
 	return s.newFakeStore(devBE)
 }
 
-func (s *deviceMgrBaseSuite) TearDownTest(c *C) {
-	s.ancillary = nil
-	s.state.Lock()
-	assertstate.ReplaceDB(s.state, nil)
-	s.state.Unlock()
-	bootloader.Force(nil)
-	dirs.SetRootDir("")
-	s.restoreGenericClassicMod()
-	s.restoreOnClassic()
-	s.restoreSanitize()
-	s.restoreCloudInitStatusRestore()
```

Comment:
This is not called anymore, is this intentional?

Reply:
yes, everything was moved to use AddCleanup from SetUpTest and in this case the BaseTest.TearDownTest is called for us so nothing to override/do